src/ima.rs

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Keylime Authors
+
+#[macro_use]
+use log::*;
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::Error;
+use std::io::SeekFrom;
+use std::path::Path;
+use std::str;
+
+use tempfile::NamedTempFile;
+
+/// IMAMeasurementList models the IMA measurement lists's last two known
+/// numbers of entries in the log and filesizes at that point
+struct ImaMeasurementList {
+    tuples: [(u64, u64); 2],
+}
+
+impl ImaMeasurementList {
+    fn reset(&mut self) -> (u64, u64) {
+        self.tuples = [(0, 0), (0, 0)];
+        return self.tuples[0];
+    }
+
+    fn update(&mut self, num_entries: u64, filesize: u64) {
+        let entry = (num_entries, filesize);
+        if entry != self.tuples[0] {
+            self.tuples = [entry, self.tuples[0]];
+        }
+    }
+
+    fn find(&mut self, nth_entry: u64) -> Option<(u64, u64)> {
+        if nth_entry >= self.tuples[0].0 {
+            return Some(self.tuples[0]);
+        }
+        if nth_entry >= self.tuples[1].0 {
+            return Some(self.tuples[1]);
+        }
+        return None;
+    }
+}
+
+// Singleton ImeMeasurementList
+static mut IMAMEASUREMENTLIST: ImaMeasurementList = ImaMeasurementList {
```

Comment:
What about moving this to [`QuoteData`](https://github.com/keylime/rust-keylime/blob/master/src/main.rs#L89) struct in main.rs, which is kept during the agent lifetime?